app/frontend/src/features/profile/edit/EditHostingPreference.test.tsx

```diff
@@ -0,0 +1,47 @@
+import {
+  render,
+  screen,
+  waitFor,
+  queryByAttribute,
+  fireEvent,
+  act,
+} from "@testing-library/react";
+import { Route, Switch } from "react-router-dom";
+import { addDefaultUser, MockedService } from "../../../test/utils";
+import wrapper from "test/hookWrapper";
+import EditHostingPreference from "./EditHostingPreference";
+import { getUser } from "test/serviceMockDefaults";
+import { service } from "service";
+import { profileRoute } from "routes";
+
+const MockProfilePage = "Mock Profile Page";
+const getUserMock = service.user.getUser as MockedService<
+  typeof service.user.getUser
+>;
+const MockProfileComponent = () => <div>{MockProfilePage}</div>;
+
+describe("EditHostingPreference", () => {
+  beforeEach(() => {
+    getUserMock.mockImplementation(getUser);
+    addDefaultUser();
+  });
+
+  it("should redirect to Profile route after successful update", async () => {
+    const { getByText } = render(
```

Comment:
```suggestion
    render(
```
Generally we don't need to do this anymore since we can use `getByText` from `screen`